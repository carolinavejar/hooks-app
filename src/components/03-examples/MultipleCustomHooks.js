import React from 'react'
import { useCounter } from '../../hooks/useCounter';
import { useFetch } from '../../hooks/useFetch'
import '../css/index.css';

export const MultipleCustomHooks = () => {

    const { counter, increment } = useCounter(1)
    const { loading, data } = useFetch(`https://www.breakingbadapi.com/api/quotes/${ counter }`);
    console.log(data);
    const {author, quote} = !!data && data[0];
    return (
        <div>
            <h1>Breaking Bad Quotes</h1>
            <hr />

            {
                loading ?
                (
                    <div className="lds-facebook"><div></div><div></div><div></div></div>
                )
                : 
                (
                    <blockquote className="blockquote float-right">
                        <p className="mb-0">{ quote }</p>
                        <hr />
                        <footer className="blockquote-footer">
                            { author }
                        </footer>
                    </blockquote>
                )
            }

            <button className="btn btn-primary" onClick={ increment }>
                Siguiente Blackquote
            </button>
           
        </div>
    )
}
