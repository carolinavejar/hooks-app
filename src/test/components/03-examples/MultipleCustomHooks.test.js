import React from 'react';
import { shallow } from 'enzyme/build';
// No las toma de setupTests.js , así que las dejaré aquí por ahora
import Enzyme from 'enzyme';
// import Adapter from 'enzyme-adapter-react-16';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { createSerializer } from 'enzyme-to-json';

Enzyme.configure({ adapter: new Adapter() });
expect.addSnapshotSerializer( createSerializer ( { mode : 'deep' } ));

import { MultipleCustomHooks } from '../../../components/03-examples/MultipleCustomHooks'
import { useFetch } from '../../../hooks/useFetch';
import { useCounter } from '../../../hooks/useCounter';


jest.mock('../../../hooks/useFetch');
jest.mock('../../../hooks/useCounter');

describe('Pruebas en useFectch', ()=> {

    beforeEach( () =>  {
        useCounter.mockReturnValue({
            counter: 10,
            increment: () => {}
        });
    });

    test('Debe mostrarse correctamente', () => {

        useFetch.mockReturnValue({
            data : null, 
            loading: true,
            error: null
        });
        const wrapper = shallow(<MultipleCustomHooks />);
        expect ( wrapper ).toMatchSnapshot();
    });

    test('Debe mostrar la información', () => {
        useFetch.mockReturnValue({
            data : [{
                author: 'Carito',
                quote: 'Hola mundo'
            }], 
            loading: false,
            error: null
        });

        const wrapper = shallow(<MultipleCustomHooks />);
        console.log(wrapper.html());
        expect(wrapper.find ('.alert').exists()).toBe(false);
        expect(wrapper.find ('.mb-0').text().trim()).toBe('Hola mundo');
        expect(wrapper.find ('footer').text().trim()).toBe('Carito');      
    })
    
})