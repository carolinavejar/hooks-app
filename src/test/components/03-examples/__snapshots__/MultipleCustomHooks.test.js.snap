// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en useFectch Debe mostrarse correctamente 1`] = `
<div>
  <h1>
    Breaking Bad Quotes
  </h1>
  <hr />
  <div
    className="lds-facebook"
  >
    <div />
    <div />
    <div />
  </div>
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Siguiente Blackquote
  </button>
</div>
`;
